include/proxy-wasm/context.h

```diff
@@ -45,14 +45,16 @@ class WasmVm;
  * @param vm_id is a string used to differentiate VMs with the same code and VM configuration.
  * @param plugin_configuration is configuration for this plugin.
  * @param fail_open if true the plugin will pass traffic as opposed to close all streams.
+ * @param key is used for caching PluginHandleBase per thread and determining whether to create a
+ * new PluginHandleBase.
```

Comment:
This sentence focuses too much on implementation details and doesn't help embedders to decide the value of the key. Perhaps something like "key is used to uniquely identify this plugin instance" would be a little more clear?